draft-ietf-quic-load-balancers.md

```diff
@@ -752,9 +773,53 @@ such as congestion window, RTT, or MTU. Opaque data MAY also allow servers to
 distinguish between retry tokens (which trigger use of certain transport
 parameters) and NEW_TOKEN frame tokens.
 
+### Token Protection with AEAD {#token-protection-with-aead}
+
+The tokens are protected using AES128-GCM as follow:
+
+* The key and IV for the retry tokens are provisioned by the configuration
+agent. The nonce, N, is formed by combining the IV with the token
+sequence number. The 64 bits of the token sequence number are left-padded
+with zeros to the size of the IV. The exclusive OR of the padded token sequence
+number and the IV forms the AEAD nonce.
+
+* The associated data is a formatted as a pseudo header as follow:
+~~~~~
+0                   1                   2                   3
+0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                                                               |
++                                                               +
+|                                                               |
++                        IP Address (128)                       +
+|                                                               |
++                                                               +
+|                                                               |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                                                               |
++                        Timestamp (64)                         +
+|                                                               |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+~~~~~
+* The input plaintext, for the AEAD is the token body.The output ciphertext
+of the AEAD is transmitted in place of token body. On the wire, the
+token is presented as:
```

Comment:
Should this make clear that the authentication tag is appended to the end of the token (assuming it is)?

Reply:
Yes it is. I copied the text from the QUIC-TLS draft, the tag is considered part of the encrypted data.

Reply:
With the "opaque" data and so on it would be helpful to show exactly where the tag goes, so as to leave no ambiguity.